nntrainer/models/neuralnet.cpp

```diff
@@ -551,15 +563,25 @@ int NeuralNetwork::train(std::vector<std::string> values) {
   status = allocate(true);
   NN_RETURN_STATUS();
 
-  /** Setup data buffer properties */
-  status = data_buffer->setClassNum(getOutputDimension()[0].width());
-  NN_RETURN_STATUS();
+  auto initiate_data_buffer = [this](std::shared_ptr<DataBuffer> &db) {
+    int status = db->setClassNum(getOutputDimension()[0].width());
```

Comment:
can we use `setLabelShape(std::vector<TensorDim>)`?

Reply:
Similar interface is being proposed. So for now it is matching with `databuffer_callback.`

Reply:
alright. Maybe add a TODO for it here.

Reply:
Done